File: backend/Watchdog.Core/Watchdog.Core.DAL/Entities/Tile.cs
using Watchdog.Core.Common.Enums.Tiles;
using Watchdog.Core.DAL.Entities.Common;

namespace Watchdog.Core.DAL.Entities
{
    public class Tile : AuditEntity<int>
    {
        public string Name { get; set; }
        public TileCategory Category { get; set; }
        public TileType Type { get; set; }
        public User User { get; set; }
        public int DashboardId { get; set; }
        public Dashboard Dashboard { get; set; }
        public string Settings { get; set; }
    }
}

File: frontend/src/app/shared/models/tile/tile.ts
import { TileCategory } from '@shared/models/tile/enums/tile-category';
import { TileType } from '@shared/models/tile/enums/tile-type';

export interface Tile {
    id: number,
    name: string,
    category: TileCategory,
    type: TileType,
    dashboardId: number,
    createdBy: number,
    createdAt: Date,
    settings: string
}
